### PR TITLE
Update docker-compose CICD with invoke commands

### PR DESCRIPTION
**Pull request by BeArchiTek**
New invoke commands to :
- generate the ENV variables from Infrahub Config with the default value (and update docker-compose file if needed)
- get the version from pyproject.toml and update docker-compose file

Rework CICD file to use the Bot token on stable branch and use the 2 new invoke commands

### CODE REVIEW

**Comment by BeArchiTek** [comment]
@ogenstad I want to update the default *_ADDRESS for the container names instead of localhost as it's our default deployment. I want your point of view on it (it seems like we override the value everywhere for now)

@fatih-acar you are using the variables in others docker files in development/ but without default + override in the services directly, so didn't change those for now

File: tasks/dev.py [hunk 1]
```diff
@@ -193,13 +193,104 @@ def migrate(context: Context, database: str = INFRAHUB_DATABASE):
...
+            var_name = var_name.strip()
+            existing_value = existing_value.strip().strip('"')
+            if var_name in env_vars:
+                default_value = env_defaults.get(var_name, None)
+                if default_value is not None:
+                    # Convert bool to lowercase strings
+                    if isinstance(default_value, bool):
+                        default_value = str(default_value).lower()
+                    # Check if the default value is an enum-like value and replace it with its actual value
+                    elif isinstance(default_value, (DatabaseType, StorageDriver, TraceExporterType, TraceTransportProtocol)):
```
Comment (fatih-acar) [inline tasks/dev.py]: I think you can directly do
```suggestion
                    elif isinstance(default_value, Enum):
```

and use `default_value.value`

File: docker-compose.yml [hunk 2]
```diff
@@ -14,97 +14,85 @@ x-infrahub-config: &infrahub_config
...
+  INFRAHUB_EXPERIMENTAL_PULL_REQUEST: "false"
+  INFRAHUB_GIT_REPOSITORIES_DIRECTORY: "repositories"
+  INFRAHUB_GIT_SYNC_INTERVAL: "10"
+  INFRAHUB_INITIAL_DEFAULT_BRANCH: "main"
+  INFRAHUB_INITIAL_ADMIN_PASSWORD: "infrahub"
+  INFRAHUB_INITIAL_ADMIN_TOKEN: "06438eb2-8019-4776-878c-0941b1f1d1ec"
   INFRAHUB_INITIAL_AGENT_PASSWORD:
-  INFRAHUB_INITIAL_AGENT_TOKEN: &agent_token "44af444d-3b26-410d-9546-b758657e026c"
-  INFRAHUB_INITIAL_DEFAULT_BRANCH:
+  INFRAHUB_INITIAL_AGENT_TOKEN:
```
Comment (fatih-acar) [inline docker-compose.yml:70]: This becomes empty, is this expected?

**Comment by ogenstad** [comment]
> @ogenstad I want to update the default *_ADDRESS for the container names instead of localhost as it's our default deployment. I want your point of view on it (it seems like we override the value everywhere for now)

For me it would make most sense to keep `localhost` as that would always be usable for local testing, however `infrahub-server` would only make sense within the context of our Docker environment. If we are to remove localhost I think it would be better to just remove it and change these values to be required. So that you have to specify the server names. Forcing users to set these values would also make sense for the SDK.

I'm not sure we should keep the INFRAHUB_INTERNAL_ADDRESS, to me this is a duplicate that is only used to set INFRAHUB_ADDRESS used by the SDK. I think the only reason to keep it is if we should continue supporting infrahub.toml files but I'd vote to get rid of those.

File: tasks/dev.py [hunk 3]
```diff
@@ -193,13 +194,145 @@ def migrate(context: Context, database: str = INFRAHUB_DATABASE):
...
+            elif not default_value_str:
+                new_config_lines.append(f"  {var}:\n")
+            else:
+                new_config_lines.append(f'  {var}: "{default_value_str}"\n')
+        elif var not in existing_vars and not default_value_str:
+            print(f"New variable {var} added")
+            new_config_lines.append(f"  {var}:\n")
+        else:
+            print(f"New variable {var} added with {default_value_str}")
+            new_config_lines.append(f'  {var}: "{default_value_str}"\n')
```
Comment (fatih-acar) [inline tasks/dev.py]:
```suggestion
            new_config_lines.append(f'  {var}: ${{{var}:-{default_value_str}}}\n')
```

File: tasks/dev.py [hunk 4]
```diff
@@ -193,13 +194,145 @@ def migrate(context: Context, database: str = INFRAHUB_DATABASE):
...
+            elif existing_value != default_value_str:
+                print(f"{var} value is different old: {existing_value} - new: {default_value_str}")
+                if not default_value_str:
+                    new_config_lines.append(f"  {var}:\n")
+                else:
+                    new_config_lines.append(f'  {var}: "{default_value_str}"\n')
+            elif not default_value_str:
+                new_config_lines.append(f"  {var}:\n")
+            else:
+                new_config_lines.append(f'  {var}: "{default_value_str}"\n')
```
Comment (fatih-acar) [inline tasks/dev.py]:
```suggestion
                new_config_lines.append(f'  {var}: ${{{var}:-{default_value_str}}}\n')
```

File: tasks/dev.py [hunk 5]
```diff
@@ -193,13 +194,145 @@ def migrate(context: Context, database: str = INFRAHUB_DATABASE):
...
+                key_name = var.replace("INFRAHUB_INITIAL_", "").lower()
+                new_config_lines.append(f'  {var}: &{key_name} "{existing_value}"\n')
+            elif default_value_str == "localhost":
+                new_config_lines.append(f"  {var}:\n")
+            elif existing_value != default_value_str:
+                print(f"{var} value is different old: {existing_value} - new: {default_value_str}")
+                if not default_value_str:
+                    new_config_lines.append(f"  {var}:\n")
+                else:
+                    new_config_lines.append(f'  {var}: "{default_value_str}"\n')
```
Comment (fatih-acar) [inline tasks/dev.py]:
```suggestion
                    new_config_lines.append(f'  {var}: ${{{var}:-{default_value_str}}}\n')
```

File: tasks/dev.py [hunk 6]
```diff
@@ -193,13 +194,145 @@ def migrate(context: Context, database: str = INFRAHUB_DATABASE):
...
+        if var in existing_vars:
+            line_idx = existing_vars[var]
+            existing_value = docker_compose[line_idx].split(":", 1)[1].strip().strip('"')
+            if existing_value.startswith("&"):
+                existing_value = existing_value.split(" ", 1)[-1].strip('"')
+
+            # Always handle special vars with anchors
+            if var in ["INFRAHUB_BROKER_USERNAME", "INFRAHUB_BROKER_PASSWORD"]:
+                key_name = var.replace("INFRAHUB_", "").lower()
+                new_config_lines.append(f'  {var}: &{key_name} "{default_value_str}"\n')
```
Comment (fatih-acar) [inline tasks/dev.py]:
```suggestion
                new_config_lines.append(f'  {var}: &{key_name} ${{{var}:-{default_value_str}}}\n')
```

File: tasks/dev.py [hunk 7]
```diff
@@ -193,13 +194,145 @@ def migrate(context: Context, database: str = INFRAHUB_DATABASE):
...
+            elif var in ["INFRAHUB_INITIAL_ADMIN_TOKEN", "INFRAHUB_INITIAL_AGENT_TOKEN"]:
+                key_name = var.replace("INFRAHUB_INITIAL_", "").lower()
+                new_config_lines.append(f'  {var}: &{key_name} "{existing_value}"\n')
```
Comment (fatih-acar) [inline tasks/dev.py]:
```suggestion
                new_config_lines.append(f'  {var}: &{key_name} ${{{var}:-{existing_value}}}\n')
```

**Review by fatih-acar — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 8]
```diff
@@ -158,6 +161,14 @@ services:
         condition: service_healthy
     environment:
       <<: *infrahub_config
+      INFRAHUB_PRODUCTION: "false"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_PRODUCTION: ${INFRAHUB_PRODUCTION:-false}
```

File: docker-compose.yml [hunk 9]
```diff
@@ -158,6 +161,14 @@ services:
...
+      INFRAHUB_LOG_LEVEL: "INFO"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_LOG_LEVEL: ${INFRAHUB_LOG_LEVEL:-INFO}
```

File: docker-compose.yml [hunk 10]
```diff
@@ -158,6 +161,14 @@ services:
...
+      INFRAHUB_BROKER_ADDRESS: "message-queue"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_BROKER_ADDRESS: ${INFRAHUB_BROKER_ADDRESS:-message-queue}
```

File: docker-compose.yml [hunk 11]
```diff
@@ -158,6 +161,14 @@ services:
...
+      INFRAHUB_CACHE_ADDRESS: "cache"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_CACHE_ADDRESS: ${INFRAHUB_CACHE_ADDRESS:-cache}
```

File: docker-compose.yml [hunk 12]
```diff
@@ -158,6 +161,14 @@ services:
...
+      INFRAHUB_DB_ADDRESS: "database"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_DB_ADDRESS: ${INFRAHUB_DB_ADDRESS:-database}
```

File: docker-compose.yml [hunk 13]
```diff
@@ -181,7 +192,15 @@ services:
       - infrahub-server
     environment:
       <<: *infrahub_config
-      INFRAHUB_API_TOKEN: *agent_token
+      INFRAHUB_PRODUCTION: "false"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_PRODUCTION: ${INFRAHUB_PRODUCTION:-false}
```

File: docker-compose.yml [hunk 14]
```diff
@@ -181,7 +192,15 @@ services:
...
+      INFRAHUB_LOG_LEVEL: "DEBUG"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_LOG_LEVEL: ${INFRAHUB_LOG_LEVEL:-DEBUG}
```

File: docker-compose.yml [hunk 15]
```diff
@@ -181,7 +192,15 @@ services:
...
+      INFRAHUB_API_TOKEN: ${INFRAHUB_INITIAL_AGENT_TOKEN:-44af444d-3b26-410d-9546-b758657e026c}
+      INFRAHUB_SECURITY_SECRET_KEY: ${INFRAHUB_SECURITY_SECRET_KEY-327f747f-efac-42be-9e73-999f08f86b92"}
+      INFRAHUB_ADDRESS: "http://infrahub-server:8000"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_ADDRESS: ${INFRAHUB_ADDRESS:-http://infrahub-server:8000}
```

File: docker-compose.yml [hunk 16]
```diff
@@ -181,7 +192,15 @@ services:
...
+      INFRAHUB_INTERNAL_ADDRESS: "http://infrahub-server:8000"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_INTERNAL_ADDRESS: ${INFRAHUB_INTERNAL_ADDRESS:-http://infrahub-server:8000}
```

File: docker-compose.yml [hunk 17]
```diff
@@ -181,7 +192,15 @@ services:
...
+      INFRAHUB_BROKER_ADDRESS: "message-queue"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_BROKER_ADDRESS: ${INFRAHUB_BROKER_ADDRESS:-message-queue}
```

File: docker-compose.yml [hunk 18]
```diff
@@ -181,7 +192,15 @@ services:
...
+      INFRAHUB_CACHE_ADDRESS: "cache"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_CACHE_ADDRESS: ${INFRAHUB_CACHE_ADDRESS:-cache}
```

File: docker-compose.yml [hunk 19]
```diff
@@ -181,7 +192,15 @@ services:
...
+      INFRAHUB_DB_ADDRESS: "database"
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_DB_ADDRESS: ${INFRAHUB_DB_ADDRESS:-database}
```

File: docker-compose.yml [hunk 20]
```diff
@@ -181,7 +192,15 @@ services:
...
       INFRAHUB_TIMEOUT: 20
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_TIMEOUT: ${INFRAHUB_TIMEOUT:-20}
```

File: docker-compose.yml [hunk 21]
```diff
@@ -181,7 +192,15 @@ services:
       - infrahub-server
     environment:
       <<: *infrahub_config
-      INFRAHUB_API_TOKEN: *agent_token
+      INFRAHUB_PRODUCTION: "false"
+      INFRAHUB_LOG_LEVEL: "DEBUG"
+      INFRAHUB_API_TOKEN: ${INFRAHUB_INITIAL_AGENT_TOKEN:-44af444d-3b26-410d-9546-b758657e026c}
+      INFRAHUB_SECURITY_SECRET_KEY: ${INFRAHUB_SECURITY_SECRET_KEY-327f747f-efac-42be-9e73-999f08f86b92"}
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_SECURITY_SECRET_KEY: ${INFRAHUB_SECURITY_SECRET_KEY:-327f747f-efac-42be-9e73-999f08f86b92"}
```

File: docker-compose.yml [hunk 22]
```diff
@@ -158,6 +161,14 @@ services:
...
     environment:
       <<: *infrahub_config
+      INFRAHUB_PRODUCTION: "false"
+      INFRAHUB_LOG_LEVEL: "INFO"
+      INFRAHUB_BROKER_ADDRESS: "message-queue"
+      INFRAHUB_CACHE_ADDRESS: "cache"
+      INFRAHUB_DB_ADDRESS: "database"
+      INFRAHUB_INITIAL_ADMIN_TOKEN: ${INFRAHUB_INITIAL_ADMIN_TOKEN:-06438eb2-8019-4776-878c-0941b1f1d1ec}
+      INFRAHUB_INITIAL_AGENT_TOKEN: ${INFRAHUB_INITIAL_AGENT_TOKEN:-44af444d-3b26-410d-9546-b758657e026c}
+      INFRAHUB_SECURITY_SECRET_KEY: ${INFRAHUB_SECURITY_SECRET_KEY-327f747f-efac-42be-9e73-999f08f86b92"}
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      INFRAHUB_SECURITY_SECRET_KEY: ${INFRAHUB_SECURITY_SECRET_KEY:-327f747f-efac-42be-9e73-999f08f86b92"}
```

**Review by fatih-acar — APPROVED** [review]

**Pull request merged**